kernel/src/process/signal/pending.rs

```diff
@@ -0,0 +1,101 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use core::sync::atomic::Ordering;
+
+use crate::{
+    prelude::*,
+    process::{
+        posix_thread::PosixThread,
+        signal::{
+            constants::SIGKILL,
+            sig_mask::{SigMask, SigSet},
+            signals::Signal,
+        },
+        Process,
+    },
+};
+
+/// Trait for handling pending signals.
+pub trait HandlePendingSignal {
+    /// Returns the thread's pending signal set.
+    ///
+    /// This includes signals that are currently blocked or ignored.
+    fn pending_signals(&self) -> SigSet;
+
+    /// Returns if there are pending signals that are neither blocked nor ignored.
+    ///
+    /// Note that ignored but not blocked signals may be dequeued silently.
+    fn has_pending(&self) -> bool;
+
+    /// Returns if a SIGKILL signal is pending.
+    fn has_pending_sigkill(&self) -> bool;
+
+    /// Dequeues the next pending signal that is not masked by `mask`.
+    ///
+    /// Returns `None` if no such signal is available.
+    fn dequeue_signal(&self, mask: &SigMask) -> Option<Box<dyn Signal>>;
+}
+
+impl HandlePendingSignal for Context<'_> {
+    fn pending_signals(&self) -> SigSet {
+        self.posix_thread.sig_queues().sig_pending() | self.process.sig_queues().sig_pending()
+    }
+
+    fn has_pending(&self) -> bool {
+        let posix_thread = self.posix_thread;
+        let process = self.process.as_ref();
+
```

Comment:
```suggestion
```

Be consistent. No need to have this blank line (`PosixThread::has_pending` does not have this)?